apps/jubmoji-quest/src/pages/quests/[id].tsx

```diff
@@ -128,25 +114,32 @@ export default function QuestDetailPage() {
             </div>
           </div>
         </QuestCard>
+
         {quest.powers.map((power) => {
           return (
             <Link key={power.id} href={`/powers/${power.id}`}>
               <PowerCard
                 title={power.name}
                 description={power.description}
                 powerType={power.powerType}
-                disabled={true} // Todo: Logic for enabling powers
+                disabled={powerIsLocked} // Todo: Logic for enabling powers
               />
             </Link>
           );
         })}
-        {quest.proofType === $Enums.ProofType.TEAM_LEADERBOARD && (
-          <div>
-            <LeaderBoard></LeaderBoard>
-            <Button variant="secondary" onClick={onUpdateTeamLeaderboardScore}>
+
+        {showLeaderBoard && (
+          <>
+            <LeaderBoard items={scoreMapping} loading={isLoadingLeaderBoard} />
+            <Button
+              variant="secondary"
+              onClick={onUpdateTeamLeaderBoardScore}
+              disabled={updateLeaderBoardMutation.isLoading}
+              loading={updateLeaderBoardMutation.isLoading}
+            >
               Update team score
```

Comment:
No need for any changes in this PR, but once https://github.com/jubmoji/jubmoji.quest/issues/67 is done, we can be more descriptive and say "Add 3 Points to Team Score" or something like that